File: frontend/lib/auth/scopes.ts
export const SCOPES = {
  SEARCH_READ: "search.read",
  DOCUMENT_READ: "document.read",
  CONNECTORS_READ: "connectors.read",
  CONNECTORS_WRITE: "connectors.write",
  ADMIN_AUDIT_READ: "admin.audit.read",
} as const

export function hasScope(scopes: string[], scope: string): boolean {
  return scopes.includes(scope)
}

/**
 * Admin-vs-member follows JWT scopes from the persisted org role.
 * Members now also get connectors.read/write for their own Google account.
 * Do not treat connectors.write as admin.
 * 
 * Owner role has same scopes as admin for JWT purposes.
 */
export function isAdmin(scopes: string[]): boolean {
  return scopes.some((scope) => scope.startsWith("admin."))
}

/**
 * Check if user has owner role (from JWT role claim, not scopes).
 * This is used for UI decisions like showing ownership transfer controls.
 */
export function isOwner(role: string | undefined): boolean {
  return role === "owner"
}

File: backend/app/services/admin/scopes.py
"""Role → JWT scope mapping for Block N admin-first RBAC."""

from __future__ import annotations

from typing import List

ADMIN_SCOPES: List[str] = [
    "search.read",
    "document.read",
    "connectors.read",
    "connectors.write",
    "admin.audit.read",
    "admin.users.read",
    "admin.users.write",
    "admin.connectors.read",
    "admin.connectors.write",
    "admin.sessions.revoke",
    "graph.admin",
    "signals.admin",
]

MEMBER_SCOPES: List[str] = [
    "search.read",
    "document.read",
    "connectors.read",
    "connectors.write",
]

VIEWER_SCOPES: List[str] = [
    "search.read",
    "document.read",
    "connectors.read",
]


def scopes_for_role(role: str) -> List[str]:
    """Return the JWT scope list for a persisted user role."""
    if role == "owner":
        return list(ADMIN_SCOPES)
    if role == "admin":
        return list(ADMIN_SCOPES)
    if role == "viewer":
        return list(VIEWER_SCOPES)
    return list(MEMBER_SCOPES)
